jwst/photom/ifu_mrs.py

```diff
@@ -0,0 +1,145 @@
+    a = timecoeff[side]['acoeff']
+    b = timecoeff[side]['bcoeff']
+    c = timecoeff[side]['ccoeff']
+    x0 = timecoeff[side]['x0']
+
+    # time depenpent function
+    corgrid = expfunc_bounded(mid_time, a, b, c, x0)
+    # now fold in the wavelength dependence
+    func = RegularGridInterpolator([binwave], corgrid,
+                                   bounds_error=False, fill_value=None)
+    return func
+
+
+def time_correction(input, detector, ftab, mid_time):
+    """
+    Find the time and wavelength dependent photom correction for MRS data.
+
+    Parameters
+    ----------
+    input : JWST data model
+        Input science data model to be corrected.
+
+    detector : string
+        MRS detector working on
+
+    ftab : stdatamodels.jwst.datamodels.MirMrsPhotomModel
+        MRS PHotom reference file
+
+    mid_time: float
+        Exposure mid time in MJD
+
+    Returns
+    -------
+    result : numpy array
+        An array of corrections to apply to data.
+    """
+
+    # Tables to be read in from MRS Photom reference file
+    # Read in the time coefficients for each channel
+    table_ch = {}
+    table_ch['ch1'] = ftab.timecoeff_ch1
+    table_ch['ch2'] = ftab.timecoeff_ch2
+    table_ch['ch3'] = ftab.timecoeff_ch3
+    table_ch['ch4'] = ftab.timecoeff_ch4
+
+    # Each MIRI MRS exposure has 2 channels
+    # We want to correct the channels seperately
+    # If we have MIRIFUSHORT data then channel 1 is on the left and channel 2
+    # is on the right. If we have MIRIFULONG  data then channel 4 is on the left
+    # and channel 3 is on right.
+
+    timecoeff = {}
+    timecoeff['left'] = {}
+    timecoeff['right'] = {}
+    left = 'ch1'
+    right = 'ch2'
+    timecoeff['left']['xstart'] = 0
+    timecoeff['left']['xend'] = 512
+    timecoeff['right']['xstart'] = 513
+    timecoeff['right']['xend'] = 1031
+    if detector == 'MIRIFULONG':
+        left = 'ch4'
+        right = 'ch3'
+        # Check the reference file has the time dependent coefficients
+        # check that table 1 wavelength bin is an array with values
+
+    # Pull out the time coefficents for the detector we are working on
+    timecoeff['left']['binwave'] = table_ch[left]['binwave']
+    timecoeff['left']['acoeff'] = table_ch[left]['acoeff']
+    timecoeff['left']['bcoeff'] = table_ch[left]['bcoeff']
+    timecoeff['left']['ccoeff'] = table_ch[left]['ccoeff']
+    timecoeff['left']['x0'] = table_ch[left]['x0']
+
+    timecoeff['right']['binwave'] = table_ch[right]['binwave']
+    timecoeff['right']['acoeff'] = table_ch[right]['acoeff']
+    timecoeff['right']['bcoeff'] = table_ch[right]['bcoeff']
+    timecoeff['right']['ccoeff'] = table_ch[right]['ccoeff']
+    timecoeff['right']['x0'] = table_ch[right]['x0']
+
+    ysize, xsize = input.data.shape
+    y, x = np.mgrid[:ysize, :xsize]
+    # the correction is time and wavelength dependent. Pull out the
+    # wavelength of the data
+    _, _, wave = input.meta.wcs(x, y)
```

Comment:
This doesn't take into account the bounding box. The preferred method is
```
from gwcs.wcstools import grid_from_bounding_box

x,y=grid_from_boundign_box(input.meta.wcs.bounding_box)
```
In addition it takes care of using the center of the pixels.

Reply:
with all of the appropriate whitespace between elements, in order to make codestyle happy!  ;-)